### PR TITLE
Document asSubscriber stage

### PR DESCRIPTION
**Pull request by raboof**
Also made sure the jdk9-only tests can actually be compiled and ran.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -102,7 +102,8 @@ lazy val akkaScalaNightly = akkaModule("akka-scala-nightly")
   .disablePlugins(ValidatePullRequest, MimaPlugin, CopyrightHeaderInPr)
 
 lazy val benchJmh = akkaModule("akka-bench-jmh")
-  .dependsOn(Seq(actor, actorTyped, stream, streamTests, persistence, distributedData, jackson, testkit).map(
+  .enablePlugins(Jdk9)
+  .dependsOn(Seq(actor, actorTyped, stream, streamTestkit, persistence, distributedData, jackson, testkit).map(
```
Comment (raboof) [inline build.sbt]: `akka-jmh-bench` depended on `akka-stream-tests`, but used only one class from it. Moved that class to the `test` configuration of `akka-stream-testkit` instead, which probably wasn't strictly necessary, but seems a bit neater at least.

**Review by johanandren — APPROVED** [review]
LGTM

I guess if we are going for apidoc for the signature we can fix that across the board later.

File: akka-docs/src/main/paradox/stream/operators/Source/asSubscriber.md [hunk 2]
```diff
@@ -4,12 +4,44 @@ Integration with Reactive Streams, materializes into a `org.reactivestreams.Subs
...
 
-TODO: We would welcome help on contributing descriptions and examples, see: https://github.com/akka/akka/issues/25646
+If you want to create a @apidoc[Source] that gets its elements from another library that supports
+[Reactive Streams](https://www.reactive-streams.org/), you can use `JavaFlowSupport.Source.asSubscriber`.
+Each time this @apidoc[Source] is materialized, it produces a materialized value of type
+@javadoc[java.util.concurrent.Flow.Subscriber](java.util.concurrent.Flow.Subscriber).
+This @javadoc[Subscriber](java.util.concurrent.Flow.Subscriber) can be attached to a
+[Reactive Streams](https://www.reactive-streams.org/) @javadoc[Publisher](java.util.concurrent.Flow.Publisher)
+to populate it.
+
```
Comment (patriknw) [inline akka-docs/src/main/paradox/stream/operators/Source/asSubscriber.md]: don't we have to clarify the difference between JDK11 `JavaFlowSupport.Source.asSubscriber` and JDK8 `Source.asSubscriber`?

Comment (raboof) [reply akka-docs/src/main/paradox/stream/operators/Source/asSubscriber.md]: Good point, added a note

**Comment by raboof** [comment]
Needs https://github.com/lightbend/paradox/pull/394 and https://github.com/lightbend/paradox/pull/395 to make the javadoc links work

**Review by patriknw — APPROVED** [review]
LGTM, so we are not merging this until the apidoc fix is in?

**Comment by raboof** [comment]
> LGTM, so we are not merging this until the apidoc fix is in?

Yes, I think this can wait. I can cut a paradox release but @ennru is looking into https://github.com/lightbend/sbt-paradox-apidoc/issues/86 for a bit first

**Comment by ennru** [comment]
The Paradox is now solved.

**Comment by raboof** [comment]
yes, verified the links

**Pull request merged**